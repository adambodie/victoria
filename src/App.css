body {
	background-color: #F5F4F0;
}

#banner {
	width: 100%;
	position: relative;
}

.title {
    display: none;
}

.nav-item {
	margin-left: 20px;
	margin-right: 20px;
}

.buttons {
	display: flex;
	justify-content: space-around;
}

.buttons div {
	width: 75px;
	height: 75px;
	border-radius: 100%;
	margin: 20px;
}
.buttons div:first-child {
	background-color: gray;
}

.buttons div:nth-child(2) {
	background-color: gray;
}

.buttons div:nth-child(3) {
	background-color: gray;
}


.content-parent {
	display: block;
	margin: 0 auto;
	width: 80%;
}
.content {
	display: flex;
	flex-wrap: wrap;
	justify-content: space-between;
}

.content div {
	width: 500px;
	height: 250px;
	background-color: gray;
	margin: 15px;
}



.photo-overlay {
	opacity: 0;
	transition: opacity .5s;
}
.photo-overlay:hover {
	opacity: 1;
}

.slide .photo-overlay,
.slide img {
	transition: transform .6s ease-out;
}

.slide .photo-overlay {
	transform: translateX(-100%);
}

.slide:hover .photo-overlay {
	transform: translateX(0);
}

.slide:hover img {
	transform: translateX(100%); 
}

h3 {
	font-size: 1.3em;
	line-height: 1.25em;
	margin-top: .85em;
	margin-bottom: .5em;
}


.photo-container {
	text-align: center;
}

.photo {
	position: relative;
	line-height: 0;
	margin-bottom: 2em;
	overflow: hidden;
}
.photo-container img {
 	max-width: 100%;
}
.facts {
	margin-top: 30px;
	display: flex;
	justify-content: space-evenly;
	flex-wrap: wrap;
}
/* ---- Photo Overlay ---- */

.photo-overlay {
	color: #fff;
	position: absolute;
	top: 0;
	right: 0;
	bottom: 0; 
	left: 0;
	padding-left: 20px;
	padding-right: 20px;
	display: flex; /* For centering text inside .photo-overlay */
	flex-direction: column;
	justify-content: center;
	align-items: center;
}

.slide .photo-overlay {
	opacity: 1;
	background: black;
}

.visit {
	margin: 20px;
	padding: 20px;
}
/*Tabs CSS
 -----------------------*/
.react-tabs__tab-list {
  border-bottom: 1px solid #aaa;
  margin: 0 0 10px;
  padding: 0;
}

.react-tabs__tab {
  display: inline-block;
  border: 1px solid transparent;
  border-bottom: none;
  bottom: -1px;
  position: relative;
  list-style: none;
  padding: 6px 12px;
  cursor: pointer;
}

.react-tabs__tab--selected {
  background: #fff;
  border-color: #aaa;
  color: black;
  border-radius: 5px 5px 0 0;
}

.react-tabs__tab--disabled {
  color: GrayText;
  cursor: default;
}

.react-tabs__tab:focus {
  box-shadow: 0 0 5px hsl(208, 99%, 50%);
  border-color: hsl(208, 99%, 50%);
  outline: none;
}

.react-tabs__tab:focus:after {
  content: "";
  position: absolute;
  height: 5px;
  left: -4px;
  right: -4px;
  bottom: -5px;
  background: #fff;
}

.react-tabs__tab-panel {
  display: none;
}

.react-tabs__tab-panel--selected {
  display: block;
}

/* ================================= 
  Media Queries
==================================== */

@media (min-width: 0) and (max-width: 768px) {
	.photo-container {
		width: 80%;
	}
}
@media (min-width: 769px) {
	.photo-container {
		width: 40%;

	}
}

@media (min-width: 480px) {
	.title {
		position: absolute;
		left: 0;
		top: 100px;
		right: 0;
		bottom: 0;
		width: 100%;
		display: flex; /* For centering text inside .photo-overlay */
		flex-direction: column;
		justify-content: center;
		align-items: center;
	}

	.title h1 {
	    text-align: center;
		font-size: 3em;
	}
	
	.buttons div {
		width: 100px;
		height: 100px;
	}	
}



