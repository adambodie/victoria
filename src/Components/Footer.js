import React, { Component } from 'react';

export default class Footer extends Component {
  render() {
	return (
		<div className="footer">
			<nav className="navbar navbar-expand-md navbar-dark bg-dark">
				<p>Adam Bodie</p>
			</nav>
	</div>
	);
	}
}
