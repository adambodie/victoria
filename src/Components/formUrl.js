const formUrl = (photoset_id, per_page) => `https://api.flickr.com/services/rest/?method=flickr.photosets.getPhotos&api_key=0c3f8d32a28de8434240115b85a28499&photoset_id=${photoset_id}&user_id=8994820%40N07&extras=tags&per_page=${per_page}&format=json&nojsoncallback=1`;

export default formUrl;