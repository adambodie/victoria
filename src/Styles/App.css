body {
	background-color: #F5F4F0;
}

button {
	display: block;
	margin: 0 auto;
	background-color: black;
	margin-bottom: 30px;
	color: white;
	padding: 20px;
}

a {
	color: white;
}
a:hover {
	color: #EE3A43;
	text-decoration:none;
}

.banner {
	width: 100%;
	position: relative;
}

.title {
    position: absolute;
    left: 0;
    top: 0;
    right: 0;
    bottom: 0;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}


.title h1 {
	font-size: 1.5em;
	color: white;
	text-shadow: 4px 4px 8px rgba(0,0,0,0.8);

}
.nav-item {
	margin-left: 20px;
	margin-right: 20px;
}

.navbar-brand {
	color: white!important;
}

.pageTitle {
	text-align: center;
}

.buttons {
	display: flex;
	justify-content: space-around;
	margin: 30px 0px;
	padding: 20px;
	box-shadow: 2px 2px 4px rgba(0,0,0, 0.5);
}


.content-parent {
	display: block;
	margin: 0 auto;
	width: 80%;
	margin-top: 30px;
}
.content {
	display: flex;
	flex-wrap: wrap;
	justify-content: space-between;
}

.visit {
	margin: 20px;
	padding: 20px;
}

.tabTitle {
	font-size: 18px;
	text-align: center;
}
/*Tabs CSS
 -----------------------*/

.react-tabs__tab-list {
  border-bottom: 1px solid #aaa;
  padding: 0;
}

.react-tabs__tab {
  display: inline-block;
  border: 1px solid transparent;
  border-bottom: none;
  bottom: -1px;
  position: relative;
  list-style: none;
  padding: 6px 12px;
  cursor: pointer;
  font-size: 10px;
}

.react-tabs__tab--selected {
  background: #fff;
  border-color: #aaa;
  color: black;
  border-radius: 5px 5px 0 0;

}

.react-tabs__tab-panel--selected {
  box-shadow: 2px 2px 4px rgba(0,0,0, 0.5);
}

.react-tabs__tab--disabled {
  color: GrayText;
  cursor: default;
}

.react-tabs__tab:focus {
  box-shadow: 0 0 5px hsl(208, 99%, 50%);
  border-color: hsl(208, 99%, 50%);
  outline: none;
}

.react-tabs__tab:focus:after {
  content: "";
  position: absolute;
  height: 5px;
  left: -4px;
  right: -4px;
  bottom: -5px;
  background: #fff;
}

.react-tabs__tab-panel {
  display: none;
}

.react-tabs__tab-panel--selected {
  display: block;
}

.homeButtons {
	text-align: center;
	width: 60px;
	height: 60px;
	display: flex;
	border-radius: 100%;
}

.notFound img{
	display: block;
	margin: 0 auto;
	padding: 20px;
}
.notFound h1 {
	text-align: center;
	margin-top: 30px;
}
/* ================================= 
  Media Queries
==================================== */



@media (min-width: 480px) {
	.tabTitle {
		font-size: 36px;
	}
	.react-tabs__tab {
	  font-size: 18px;
	}
	.usd, .cad {
		width: 45%;
	}	
}


