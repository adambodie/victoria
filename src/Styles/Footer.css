footer {
	display: none;
}

.buttons {
	display: flex;
	justify-content: space-around;
	position: fixed;
	bottom: 0;
	z-index: 1000;
	background: #F5F4F0;
	margin-top: -10px;
	padding: 20px;
	box-shadow: 2px 2px 4px rgba(0,0,0, 0.5);
	width: 100%;
}

@media (min-width: 512px) {
	footer {
		display: block;
	}
	nav p {
		color: white;
	}
	.buttons {
		display: none;
	}
}
