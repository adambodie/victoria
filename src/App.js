import React, { Component } from 'react';
import {
  HashRouter as Router,
  Route,
  Switch
} from 'react-router-dom';
import './Styles/App.css';
import Header from './Components/Header';
import HomeRoute from './Components/Routes/HomeRoute';
import PhotoRoute from './Components/Routes/PhotoRoute';
import FactRoute from './Components/Routes/FactRoute';
import BuyRoute from './Components/Routes/BuyRoute';
import RestaurantRoute from './Components/Routes/RestaurantRoute';
import NotFoundRoute from './Components/Routes/NotFoundRoute';
import Footer from './Components/Footer';

class App extends Component {
  render() {
    return (
      	<Router>
			<div className="App">
				<Header/>
					<Switch>
						<Route exact path = "/" render={() => <HomeRoute />}/>
						<Route path = "/visit" render={() => <PhotoRoute />}/>
						<Route path = "/eat" render={() => <RestaurantRoute />}/>
						<Route path = "/learn" render={() => <FactRoute />}/>
						<Route path = "/buy" render={() => <BuyRoute />}/>
						<Route component={NotFoundRoute} />
					</Switch>
				<Footer/>
			</div>
      </Router>
    );
  }
}

export default App;
