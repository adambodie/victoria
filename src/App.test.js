import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Footer from './Components/Footer';
import renderer from 'react-test-renderer';
import formUrl from './Components/formUrl.js';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correct url', () => {
  const tree = renderer.create(formUrl('135847282452394', '25')).toJSON();
  expect(tree).toMatchSnapshot();
});
