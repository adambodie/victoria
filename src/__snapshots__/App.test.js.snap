// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct url 1`] = `"https://api.flickr.com/services/rest/?method=flickr.photosets.getPhotos&api_key=0c3f8d32a28de8434240115b85a28499&photoset_id=135847282452394&user_id=8994820%40N07&extras=tags&per_page=25&format=json&nojsoncallback=1"`;
